### PR TITLE
Text Box functionality

### LINKED ISSUE #2: Implement Demo QA website - Elements -> Text Box feature
Implement automation for the Demo QA website's Elements > Text Box feature.

**Task Details:**
- Automate testing for the Text Box functionality in the Elements section
- Ensure form fields (Full Name, Email, Current Address, Permanent Address) are correctly filled and submitted
- Validate output values after submission
- Handle positive and negative test cases for the Text Box form
- Add clear assertions and error handling to ensure test reliability

**Acceptance Criteria:**
- Automated tests for all input fields
- Output validation matches submitted data
- Edge cases (invalid email, empty fields) are handled
- Tests are integrated with the project's test suite

Refer to the [Demo QA Elements Text Box page](https://demoqa.com/text-box) for feature reference.

### PR DESCRIPTION
**Pull request by Umashankar-Agarwal**
"This PR implements the Text Box automation for the Elements section. Closes #2 ."

## Summary by CodeRabbit

* **New Features**
  * Added data-driven Text Box test coverage with a new end-to-end test and corresponding page object.
  * Added a reusable test data provider for form inputs.

* **Improvements**
  * Introduced centralized logging and a runtime logging configuration for clearer test output.
  * Enabled parameterized test URL support for flexible environments.
  * Added build-time support for annotation-based tooling to reduce boilerplate.

### CODE REVIEW

File: src/main/java/pages/elements/TextBoxPage.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.PageFactory;
+import pages.Consonants;
+import utils.Reusuables;
+
+public class TextBoxPage extends CardAccessPage {
+
+    private final WebDriver driver;
+
+
```
Comment (Umashankar-Agarwal) [inline src/main/java/pages/elements/TextBoxPage.java]: remove the spaces
